vcd/datasource_vcd_lb_service_monitor.go

```diff
@@ -0,0 +1,90 @@
+package vcd
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+)
+
+func datasourceVcdLbServiceMonitor() *schema.Resource {
+	return &schema.Resource{
+		Read: datasourceVcdLbServiceMonitorRead,
+		Schema: map[string]*schema.Schema{
+			"vdc": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"org": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"edge_gateway": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Edge gateway name",
+			},
+			"name": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Service Monitor name",
+			},
+			"interval": &schema.Schema{
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"timeout": &schema.Schema{
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"max_retries": &schema.Schema{
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"type": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"expected": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"method": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"url": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"send": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"receive": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"extension": {
+				Type:     schema.TypeMap,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func datasourceVcdLbServiceMonitorRead(d *schema.ResourceData, meta interface{}) error {
```

Comment:
It's behaving about IDs a bit differently.

Reply:
Yes, it's only one line difference and yet, at the moment didn't find easy solution to avoid that. Only one read function and id set and flatten in wrapper functions.

Reply:
It's actually even more than ID. It also includes different search patterns (`readLBMonitor, err := edgeGateway.ReadLBServiceMonitor(&types.LBMonitor{Name: d.Get("name").(string)})` searches by Name, inside resource it searched by ID.

Reply:
ok. Let's leave it as it is then